engine/src/main/scala/cromwell/engine/workflow/WorkflowDockerLookupActor.scala

```diff
@@ -104,14 +111,27 @@ class WorkflowDockerLookupActor private[workflow](workflowId: WorkflowId,
 
   whenUnhandled {
     case Event(DockerHashActorTimeout(request), data) =>
-      val reason = new Exception(s"Timeout looking up docker hash")
-      data.hashRequests(request) foreach { _ ! WorkflowDockerLookupFailure(reason, request) }
-      val updatedData = data.copy(hashRequests = data.hashRequests - request)
-      stay() using updatedData
+      val reason = new Exception(s"Timeout looking up hash for Docker image ${request.dockerImageID} in state $stateName")
+      data.hashRequests.get(request.dockerImageID) match {
+        case Some(requestsAndReplyTos) =>
+          requestsAndReplyTos foreach { case RequestAndReplyTo(_, replyTo) =>
+            replyTo ! WorkflowDockerLookupFailure(reason, request)
+          }
+          val updatedData = data.copy(hashRequests = data.hashRequests - request.dockerImageID)
+          stay() using updatedData
+        case None =>
+          val headline = s"Unable to find requesters for timed out lookup of Docker image '${request.dockerImageID}' in state $stateName"
+          val pendingImageIdsAndCounts = stateData.hashRequests.toList map { case (imageId, requestAndReplyTos) => s"$imageId -> ${requestAndReplyTos.size}" }
+          val message = pendingImageIdsAndCounts.mkString(headline + "\n" + "Pending image ID requests with requester counts: ", ", ", "")
+          fail(new RuntimeException(message) with NoStackTrace)
+      }
     case Event(TransitionToFailed(cause), data) =>
       log.error(cause, s"Workflow Docker lookup actor for $workflowId transitioning to Failed")
       val updatedData = respondToAllRequestsWithTerminalFailure(FailedException, data)
       goto(Terminal) using updatedData.withFailureCause(FailedException)
+    case Event(message, _) =>
+      log.warning(s"Unexpected message $message in $stateName state")
```

Comment:
TOL: I would consider this an `error("Programmer Error: ...")` rather than a warn... but I'll leave that up to you...